packages/access-client/src/agent-data.js

```diff
@@ -152,9 +168,7 @@ export class AgentData {
   sessionProof() {
     for (const { delegation } of this.delegations.values()) {
       const cap = delegation.capabilities.find(
-        (c) =>
-          // @ts-expect-error "key" does not exist in object, unless it's a session capability
-          c.can === Access.session.can && c.nb?.key === this.principal.did()
+        (c) => c.can === Access.session.can // TODO we should make sure this is the current session proof - we were checking nb.key but that doesn't seem to exist in the staging ucan/attest at the moment
```

Comment:
it's also possible to do something like `Access.session.match(...).is(...)` and it's a type guard in a nice way, and it might be useful here (or not)